src/solve.c

```diff
@@ -130,3 +132,56 @@ void run(void)
 			return;
 	}
 }
+
+/**
+ * @brief Function to save the maze sequence on EEPROM.
+ */
+void save_maze(void)
+{
+	uint32_t save_status = 0;
+
+	save_status = eeprom_flash_page(FLASH_EEPROM_ADDRESS_MAZE,
+					(uint8_t *)run_sequence, MAZE_AREA);
+
+	if (save_status != RESULT_OK)
+		LOG_ERROR("EEPROM save error %" PRIu32, save_status);
+}
+
+/**
+ * @brief Function to load the maze sequence from EEPROM to static on RAM.
+ */
+void load_maze(void)
+{
+	eeprom_read_data(FLASH_EEPROM_ADDRESS_MAZE, MAZE_AREA,
+			 (uint8_t *)run_sequence);
+}
+
+/**
+ * @brief Function to reset the maze sequence on EEPROM.
+ */
+void reset_maze(void)
+{
+	uint32_t erase_status = 0;
+
+	erase_status = eeprom_erase_page(FLASH_EEPROM_ADDRESS_MAZE);
+	if (erase_status != RESULT_OK)
+		LOG_ERROR("EEPROM reset error %" PRIu32, erase_status);
+}
+
+/**
+ * @brief Function to check if the maze sequence is saved on EEPROM.
+ *
+ *@return bool
+ */
+bool maze_is_saved(void)
+{
+	uint8_t *byte_check = 0;
```

Comment:
I think this is incorrect? At least it seems to be broken (i.e.: this function does not run)

You do not even need that initialization anyway. :wink: